Tests/Unit/Hooks/HookProviderTest.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OliverKlee\Seminars\Tests\Unit\Hooks;
+
+use Nimut\TestingFramework\TestCase\UnitTestCase;
+use OliverKlee\Seminars\Hooks\HookProvider;
+use OliverKlee\Seminars\Hooks\Interfaces\Hook;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookInterface;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookInterfaceNotExtendsHook;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookImplementor;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookImplementor2;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookImplementorNotImplementsInterface;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Test case.
+ *
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+class HookProviderTest extends UnitTestCase
+{
+    /**
+     * @var array
+     */
+    private $extConfBackup = [];
+
+    protected function setUp()
+    {
+        $this->extConfBackup = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'];
+    }
+
+    protected function tearDown()
+    {
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'] = $this->extConfBackup;
+
+        parent::tearDown();
+    }
+
+    /*
+     * Utility functions
+     */
+
+    /**
+     * Creates a TestingHookInterface implementor object
+     *
+     * @return TestingHookImplementor
+     */
+    private function createTestingHookImplementor(): TestingHookImplementor
+    {
+        return new TestingHookImplementor();
+    }
+
+    /**
+     * Creates a second TestingHookInterface implementor object
+     *
+     * @return TestingHookImplementor2
+     */
+    private function createTestingHookImplementor2(): TestingHookImplementor2
+    {
+        return new TestingHookImplementor2();
+    }
+
+    /**
+     * Creates a TestingHookInterface accepting Hook object
+     *
+     * @param string $index
+     *
+     * @return HookProvider
+     */
+    private function createHookObject(string $index = ''): HookProvider
+    {
+        if ($index === '') {
+            return new HookProvider(TestingHookInterface::class);
+        }
+
+        return new HookProvider(TestingHookInterface::class, $index);
```

Comment:
As the default value of the `$index` parameter in the `HookProvider` constructor is an empty string, this condition is not necessary.

Reply:
… and then maybe we don't need this method after all. (Your choice … whatever your prefer here.)

Reply:
In fact I intended to make sure to omit the parameter completely so changing the default value in the class will make the tests fail.

Reply:
Are you going to change anything here, or is the current state ready for review?

Reply:
If you are OK with keeping the state, this is what I would prefer. Thanks for agreeing on going less `private`.